server/src/main/java/org/opensearch/search/aggregations/support/MissingValues.java

```diff
@@ -365,6 +366,9 @@ public int docValueCount() {
             @Override
             public long nextOrd() throws IOException {
                 if (hasOrds) {
+                    if (++ordCount > values.docValueCount()) {
```

Comment:
@msfroh This is failing due to the check we are doing here. 

The `values` here is `MultiOrdinals$MultiOrds` ( for `TermsAggregatorTests #testBucketInTermsAggregationWithMissingValue ` ) and its `docValueCount` decreases with each call to `nextOrd` 

```
 public int docValueCount() {
      return Math.toIntExact(currentEndOffset - currentOffset);
}
```

Reply:
I don't think this will be an issue. WDYT ?

Reply:
Hmm... that implementation in `MultiDocs` seems wrong. The response for `docValueCount()` should only change when we move to another doc. If it's changing as a result of `nextOrd` calls, that's wrong.